pkg/asset/manifests/clusterapi/cluster.go

```diff
@@ -0,0 +1,229 @@
+	"github.com/openshift/installer/pkg/asset/ignition/machine"
+	"github.com/openshift/installer/pkg/asset/installconfig"
+	"github.com/openshift/installer/pkg/asset/manifests"
+	"github.com/openshift/installer/pkg/asset/manifests/aws"
+	"github.com/openshift/installer/pkg/asset/manifests/azure"
+	"github.com/openshift/installer/pkg/asset/manifests/capiutils"
+	"github.com/openshift/installer/pkg/asset/openshiftinstall"
+	"github.com/openshift/installer/pkg/asset/rhcos"
+	"github.com/openshift/installer/pkg/clusterapi"
+)
+
+const (
+	manifestDir = "cluster-api"
+)
+
+var _ asset.WritableRuntimeAsset = (*Cluster)(nil)
+
+// Cluster generates manifests for target cluster
+// creation using CAPI.
+type Cluster struct {
+	FileList []*asset.RuntimeFile
+}
+
+// Name returns a human friendly name for the operator.
+func (c *Cluster) Name() string {
+	return "Cluster API Manifests"
+}
+
+// Dependencies returns all of the dependencies directly needed by the
+// ClusterAPI asset.
+func (c *Cluster) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&installconfig.InstallConfig{},
+		&installconfig.ClusterID{},
+		&openshiftinstall.Config{},
+		&manifests.FeatureGate{},
+		&bootstrap.Bootstrap{},
+		&machine.Master{},
+		new(rhcos.Image),
+	}
+}
+
+// Generate generates the respective operator config.yml files.
+func (c *Cluster) Generate(dependencies asset.Parents) error {
+	installConfig := &installconfig.InstallConfig{}
+	clusterID := &installconfig.ClusterID{}
+	openshiftInstall := &openshiftinstall.Config{}
+	featureGate := &manifests.FeatureGate{}
+	bootstrapIgnAsset := &bootstrap.Bootstrap{}
+	masterIgnAsset := &machine.Master{}
+	rhcosImage := new(rhcos.Image)
+	dependencies.Get(installConfig, clusterID, openshiftInstall, bootstrapIgnAsset, masterIgnAsset, featureGate, rhcosImage)
+
+	// If the feature gate is not enabled, do not generate any manifests.
+	if !capiutils.IsEnabled(installConfig) {
+		return nil
+	}
+
+	c.FileList = []*asset.RuntimeFile{}
+
+	namespace := &corev1.Namespace{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: capiutils.Namespace,
+		},
+	}
+	c.FileList = append(c.FileList, &asset.RuntimeFile{Object: namespace, File: asset.File{Filename: "00_capi-namespace.yaml"}})
+
+	cluster := &clusterv1.Cluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterID.InfraID,
+			Namespace: capiutils.Namespace,
+		},
+		Spec: clusterv1.ClusterSpec{},
+	}
+	c.FileList = append(c.FileList, &asset.RuntimeFile{Object: cluster, File: asset.File{Filename: "01_capi-cluster.yaml"}})
+
+	// Gather the ignition files, and store them in a secret.
+	{
+		masterIgn := string(masterIgnAsset.Files()[0].Data)
+		bootstrapIgn, err := injectInstallInfo(bootstrapIgnAsset.Files()[0].Data)
+		if err != nil {
+			return errors.Wrap(err, "unable to inject installation info")
+		}
+		c.FileList = append(c.FileList,
+			&asset.RuntimeFile{
+				File: asset.File{Filename: "01_ignition-secret-master.yaml"},
```

Comment:
The manifests are not created in the `cluster-api` dir because we never prepend the dir to the filename. A couple of ideas, we could either: 

- prepend the dir each time we create a filename: 
https://github.com/openshift/installer/blob/c7cedf047cd9431831258fda0d7d443befa9bf9d/pkg/asset/manifests/operators.go#L27
- or, at the end of the generate function, iterate through all the files and prepend the manifest dir for all files